packages/flutter_tools/lib/src/analyze_project_validator.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'analyze_project.dart';
+import 'project.dart';
+
+abstract class ProjectValidatorTask {
```

Comment:
Nit: name this `ProjectValidator`. Note that this is more like a `DoctorValidator` than a `ValidatorTask` (which is a convenience class that wraps both a `DoctorValidator` and a `Future<ValidationResult>`.

```suggestion
abstract class ProjectValidator {
```

Reply:
Also, I believe this can be made `const`.